src/flows_dst.rs

```diff
@@ -669,6 +678,78 @@ impl<'a> ProcessMerge<'a> {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub struct ProcessSplit<'a>(pub &'a mut MemberState);
+
+impl<'a> ProcessSplit<'a> {
+    pub fn start_event_loop(&mut self) {
+        self.mut_routine_state().is_active = true;
+        self.vote_for_section_split();
+    }
+
+    fn exit_event_loop(&mut self) {
+        self.mut_routine_state().is_active = false;
+        self.0
+            .as_start_merge_split_and_change_elders()
+            .transition_exit_process_split()
+    }
+
+    pub fn try_next(&mut self, event: WaitedEvent) -> TryResult {
+        match event {
+            WaitedEvent::ParsecConsensus(vote) => self.try_consensus(&vote),
+            _ => TryResult::Unhandled,
+        }
+    }
+
+    fn try_consensus(&mut self, vote: &ParsecVote) -> TryResult {
+        if !self.routine_state().wait_votes.contains(&vote) {
+            return TryResult::Unhandled;
+        }
+
+        let wait_votes = &mut self.mut_routine_state().wait_votes;
+        wait_votes.retain(|wait_vote| wait_vote != vote);
+
+        if wait_votes.is_empty() {
+            self.complete_split();
+            self.mark_elder_change();
+            self.exit_event_loop();
+        }
+        TryResult::Handled
+    }
+
+    fn vote_for_section_split(&mut self) {
+        let votes = self.0.action.get_section_split_votes();
+        self.mut_routine_state().wait_votes = votes;
+
+        for vote in &self.routine_state().wait_votes {
+            self.0.action.vote_parsec(*vote);
+        }
+    }
+
+    fn routine_state(&self) -> &ProcessSplitState {
+        &self
+            .0
+            .start_merge_split_and_change_elders
+            .sub_routine_process_split
+    }
+
+    fn mut_routine_state(&mut self) -> &mut ProcessSplitState {
```

Comment:
I think the convention used in the rust ecosystem is to put "mut" at the end, that is: `routing_state_mut`, not `mut_routing_state`. Example: `RefCell::borrow_mut`, `Vec::get_mut`, `slice::last_mut`, ...

Reply:
Ah good point. I think this would touch the whole crate so I'll do this as a follow up tidy PR